backend/app/database/exceptions.py

```diff
@@ -0,0 +1,8 @@
+from abc import ABC, abstractmethod
+
+
+class DatabaseException(ABC, BaseException):
+    @property
+    @abstractmethod
+    def message(self) -> str:
```

Comment:
You don't really need abstract message on exceptions, it's baked into the exception system already.